src/persist/src/indexed/columnar.rs

```diff
@@ -283,11 +289,26 @@ impl<'a> ColumnarRecordsRef<'a> {
         if idx >= self.len {
             return None;
         }
+
+        let key_start = self.key_offsets[idx]
```

Comment:
Ooc what type is it? We could potentially also use CastFrom here

Reply:
I was hoping we could do the same, unfortunately it's an `i32` :/